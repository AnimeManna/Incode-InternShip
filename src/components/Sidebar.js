import React, {Component} from 'react'

import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
    toolbar:theme.mixins.toolbar

})

class Sidebar extends Component{
    render(){
        const classes = this.props.classes;
        return(
            <div>
                Hello sidebar
            </div>
        )
    }
}

export default withStyles(styles)(Sidebar)