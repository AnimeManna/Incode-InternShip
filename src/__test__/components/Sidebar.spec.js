import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Sidebar from '../../components/Sidebar';

describe('<Sidebar />', () => {
  const props = {
    categories: [
      {
        title: 'Test',
        id: 1,
        description: 'Test',
      },
      {
        title: 'Test2',
        id: 2,
        description: 'Test2',
      },
      {
        title: 'Test3',
        id: 3,
        description: 'Test3',
      },
    ],
    isAuth: true,
    snackBarMessage: 'Успешный тест',
    snackBarStatus: true,
    getIsLoaded: true,
  };
  describe(' render() ', () => {
    test('Render the component', () => {
      const {
        categories, isAuth, snackBarMessage, snackBarStatus, getIsLoaded,
      } = props;
      const wrapper = shallow(
        <Sidebar
          isAuth={isAuth}
          categories={categories}
          snackBarStatus={snackBarStatus}
          snackBarMessage={snackBarMessage}
          getIsLoaded={getIsLoaded}
        />,
      );
      const component = wrapper.dive();
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
