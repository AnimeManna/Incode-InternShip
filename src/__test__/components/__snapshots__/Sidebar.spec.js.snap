// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar />  render()  Render the component 1`] = `
<div
  className="Sidebar-Sidebar-1"
>
  <WithStyles(List)
    component="nav"
  >
    <Link
      className="Sidebar-Sidebar__Button-2"
      key="1"
      replace={false}
      to="/home"
    >
      <WithStyles(ListItem)
        button={true}
      >
        <div
          className="Sidebar-Sidebar__ItemText-3"
          onClick={[Function]}
        >
          <WithStyles(ListItemIcon)>
            <pure(ChatIcon) />
          </WithStyles(ListItemIcon)>
          <WithStyles(ListItemText)
            inset={true}
            primary="Test"
          />
        </div>
        <WithStyles(ListItemIcon)
          onClick={[Function]}
        >
          <pure(DeleteIcon) />
        </WithStyles(ListItemIcon)>
      </WithStyles(ListItem)>
      <WithStyles(Divider) />
    </Link>
    <Link
      className="Sidebar-Sidebar__Button-2"
      key="2"
      replace={false}
      to="/home"
    >
      <WithStyles(ListItem)
        button={true}
      >
        <div
          className="Sidebar-Sidebar__ItemText-3"
          onClick={[Function]}
        >
          <WithStyles(ListItemIcon)>
            <pure(ChatIcon) />
          </WithStyles(ListItemIcon)>
          <WithStyles(ListItemText)
            inset={true}
            primary="Test2"
          />
        </div>
        <WithStyles(ListItemIcon)
          onClick={[Function]}
        >
          <pure(DeleteIcon) />
        </WithStyles(ListItemIcon)>
      </WithStyles(ListItem)>
      <WithStyles(Divider) />
    </Link>
    <Link
      className="Sidebar-Sidebar__Button-2"
      key="3"
      replace={false}
      to="/home"
    >
      <WithStyles(ListItem)
        button={true}
      >
        <div
          className="Sidebar-Sidebar__ItemText-3"
          onClick={[Function]}
        >
          <WithStyles(ListItemIcon)>
            <pure(ChatIcon) />
          </WithStyles(ListItemIcon)>
          <WithStyles(ListItemText)
            inset={true}
            primary="Test3"
          />
        </div>
        <WithStyles(ListItemIcon)
          onClick={[Function]}
        >
          <pure(DeleteIcon) />
        </WithStyles(ListItemIcon)>
      </WithStyles(ListItem)>
      <WithStyles(Divider) />
    </Link>
  </WithStyles(List)>
  <WithStyles(Snackbar)
    ContentProps={
      Object {
        "aria-describedby": "message-id",
      }
    }
    anchorOrigin={
      Object {
        "horizontal": "left",
        "vertical": "bottom",
      }
    }
    message={
      <span
        id="message-id"
      >
        Успешный тест
      </span>
    }
    open={true}
  />
</div>
`;

exports[`<Sidebar />  render()  Render the component with fail 1`] = `
<div
  className="Sidebar-Sidebar-1"
>
  <WithStyles(List)
    component="nav"
  >
    <Link
      className="Sidebar-Sidebar__Button-2"
      key="1"
      replace={false}
      to="/home"
    >
      <WithStyles(ListItem)
        button={true}
      >
        <div
          className="Sidebar-Sidebar__ItemText-3"
          onClick={[Function]}
        >
          <WithStyles(ListItemIcon)>
            <pure(ChatIcon) />
          </WithStyles(ListItemIcon)>
          <WithStyles(ListItemText)
            inset={true}
            primary="Test"
          />
        </div>
        <WithStyles(ListItemIcon)
          onClick={[Function]}
        >
          <pure(DeleteIcon) />
        </WithStyles(ListItemIcon)>
      </WithStyles(ListItem)>
      <WithStyles(Divider) />
    </Link>
    <Link
      className="Sidebar-Sidebar__Button-2"
      key="2"
      replace={false}
      to="/home"
    >
      <WithStyles(ListItem)
        button={true}
      >
        <div
          className="Sidebar-Sidebar__ItemText-3"
          onClick={[Function]}
        >
          <WithStyles(ListItemIcon)>
            <pure(ChatIcon) />
          </WithStyles(ListItemIcon)>
          <WithStyles(ListItemText)
            inset={true}
            primary="Test2"
          />
        </div>
        <WithStyles(ListItemIcon)
          onClick={[Function]}
        >
          <pure(DeleteIcon) />
        </WithStyles(ListItemIcon)>
      </WithStyles(ListItem)>
      <WithStyles(Divider) />
    </Link>
    <Link
      className="Sidebar-Sidebar__Button-2"
      key="3"
      replace={false}
      to="/home"
    >
      <WithStyles(ListItem)
        button={true}
      >
        <div
          className="Sidebar-Sidebar__ItemText-3"
          onClick={[Function]}
        >
          <WithStyles(ListItemIcon)>
            <pure(ChatIcon) />
          </WithStyles(ListItemIcon)>
          <WithStyles(ListItemText)
            inset={true}
            primary="Test3"
          />
        </div>
        <WithStyles(ListItemIcon)
          onClick={[Function]}
        >
          <pure(DeleteIcon) />
        </WithStyles(ListItemIcon)>
      </WithStyles(ListItem)>
      <WithStyles(Divider) />
    </Link>
  </WithStyles(List)>
  <WithStyles(Snackbar)
    ContentProps={
      Object {
        "aria-describedby": "message-id",
      }
    }
    anchorOrigin={
      Object {
        "horizontal": "left",
        "vertical": "bottom",
      }
    }
    message={
      <span
        id="message-id"
      >
        Успешный тест
      </span>
    }
    open={true}
  />
</div>
`;
